src/main/java/org/opensearch/knn/quantization/models/quantizationParams/QuantizationParams.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationParams;
+
+import org.opensearch.knn.quantization.enums.QuantizationType;
+
+import java.io.Serializable;
+
+/**
+ * Interface for quantization parameters.
+ * This interface defines the basic contract for all quantization parameter types.
+ * It provides methods to retrieve the quantization type and a unique type identifier.
+ * Implementations of this interface are expected to provide specific configurations
+ * for various quantization strategies.
+ */
+public interface QuantizationParams extends Serializable {
```

Comment:
Question: Just extending the Serializable will that be sufficient? I think we should add the functions writeObject/readObject to ensure that we are controlling how the serializing is happening for the param class.

Example:
```
// old
Param{
   var1 => abc
   var2 => def
}

// new code
Param{
   var2 => bbb
   var1 => aaa
   var3 => asdfasdf
}
```

Reply:
+1 this should help us manage backward incompatible changes